core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -45,7 +45,7 @@
  * Base class for {@link TableScan} implementations.
  */
 @SuppressWarnings("checkstyle:OverloadMethodsDeclarationOrder")
-abstract class BaseTableScan implements TableScan {
+public abstract class BaseTableScan implements TableScan {
```

Comment:
We currently need this to be public ignorer to extract table operations and other table settings in the PlanScanAction which doesn't live in the same package.

Reply:
What parts need to be accessed that aren't part of the public API? Should we just add those to the `TableScan` API?

Reply:
TableOps and Options